lib/fizzy/execute.cpp

```diff
@@ -840,8 +859,9 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
                     throw std::bad_alloc();
```

Comment:
Should this use `std::bad_alloc` or `std::length_error` ?

Reply:
The `bad_alloc` is better but not perfect fit out of these two.

Reply:
Actually, here I'd see another refactoring: do not use `throw`.
```
uint32_t ret = (new_pages <= instance.memory_pages_limit) ? static_cast<uint32_t>(cur_pages) : static_cast<uint32_t>(-1);
try {
  memory->resize(new_pages * PageSize);
} catch (...) { ... }

Reply:
It shouldn't go to `resize` though, if it's over the limit.
I can make a version with `if`, but it looks somewhat uglier.

Reply:
> The `bad_alloc` is better but not perfect fit out of these two.

Why it it better and what's the "fit" here? I'd say requirement for `memory.grow` to fail in case requested size is out of bounds is similar to requirement for `basic_string` to throw `length_error` in case `count > max_size()`.
`bad_alloc` represents failure of some lower level allocation mechanism. 

But I think in practice it doesn't matter and I'm fine with either.